src/h5web/visualizations/index.tsx

```diff
@@ -151,36 +157,38 @@ export const VIS_DEFS: Record<Vis, VisDef> = {
     Container: NxImageContainer,
     supportsEntity: (entity, metadata) => {
       const group = getNxDataGroup(entity, metadata);
-
       if (!group) {
-        return false;
+        return false; // entity is not a group and doesn't have a `default` attribute
       }
 
       const signal = getAttributeValue(group, 'signal');
-      if (typeof signal !== 'string') {
-        return false;
-      }
+      assertDefined(signal, "Expected 'signal' attribute");
+      assertStr(signal, "Expected 'signal' attribute to be a string");
 
       const dataset = getLinkedEntity(signal, group, metadata);
-      if (
-        !dataset ||
-        !isDataset(dataset) ||
-        !isNumericType(dataset.type) ||
-        !isSimpleShape(dataset.shape)
-      ) {
-        return false;
-      }
+      assertDefined(dataset, `Expected "${signal}" signal entity to exist`);
+      assertDataset(dataset, `Expected "${signal}" signal to be a dataset`);
+
+      assertNumericType(
+        dataset.type,
+        `Expected "${signal}" dataset to have numeric type`
+      );
+      assertSimpleShape(
+        dataset.shape,
+        `Expected "${signal}" dataset to have simple shape`
+      );
 
       const dimsCount = dataset.shape.dims.length;
       if (dimsCount < 2) {
         return false;
```

Comment:
I don't throw if `dimsCount` equals `0` like `NXSpectrum`'s supportsEntity` does. Again, not a fan of doing things differently here. I'd like the two functions to be exactly the same apart from the `interpretation` equality. Again, I'll refactor this later to not clutter this PR.